### ISSUE TITLE
Logging revamp.

### ISSUE THREAD

**Issue by LexManos** [dzikoysk/reposilite#450]
Currently if you're an active server the CLI is a nightmare to work with.
I would propose a revamp of the logging system to address a few things:
Logging to a file:
  This would just be nice to contain ALL errors and debug data in a way we can easily download and parse elseware.
  This file should be cycled daily/hourly to prevent it eating a ton of space.
  
 Limiting what's piped to dashboard/cli:
   On an active server {getting thousands of requests a second} the CLI is impossible to sanely use.
   So my suggestion is to add labels to printed things, and filter out only requested things in the cli.
   Potentially either turn it into a separate tab that filters things out client side, or adding a command to toggle specifc filters.
   Basically I want to be able to use the cli on an active server without seeing requests/errors

**Comment by dzikoysk** [comment]
Yeah, I've also noticed this issue in my repository and it's definitely something that must be improved. In general, I think it could be implemented with #444.

**Comment by dzikoysk** [comment]
Ok, I've made a tiny library based on a code from my another project:

* https://github.com/dzikoysk/dynamic-logger

We can now perform various operations and we're not limited to tinylog possibilities. We can also get rid of [custom writer](https://github.com/dzikoysk/reposilite/blob/master/reposilite-backend/src/main/resources/META-INF/services/org.tinylog.writers.Writer) and simplify tests that relay on this mechanism. 

I wonder if we want to only dynamically switch between logging levels, e.g.:
```
$ level DEBUG
// cli now prints everything
$ level INFO
// reduced log
```
Or do we really want to provide some kind of advanced querying 🤔 I'd limit runtime querying possibilities to reduce the overhead caused by log storage.

**Comment by LexManos** [comment]
I would say if you want to expand runtime logging in the CLI window, then only make it for new messages. Admins should be able to access the logs folder and get any previous messages. 

The other option is to make the CLI ONLY commands/responses no way to change the logger level. But add a new log tab that shows everything, but allows for toggling different levels on/off with a checkbox. Something like the Minecraft Launcher's logger: (ignore the horrible colors)
![](https://i.imgur.com/C2lKGKV.png)

**Comment by dzikoysk** [comment]
Available UI filters in 3.x:

https://user-images.githubusercontent.com/4235722/131699993-f02251a5-0394-42b9-8b45-16595fd5124b.mp4
